substrate/frame/migrations/src/lib.rs

```diff
@@ -700,6 +730,16 @@ impl<T: Config> Pallet<T> {
 		}
 
 		let max_steps = T::Migrations::nth_max_steps(cursor.index);
+
+		// If this is the first time running this migration, exec the pre-upgrade hook.
+		#[cfg(feature = "try-runtime")]
+		if !PreUpgradeBytes::<T>::contains_key(&bounded_id) {
+			let bytes = T::Migrations::nth_pre_upgrade(cursor.index)
+				.expect("Invalid cursor.index")
+				.expect("Pre-upgrade failed");
+			PreUpgradeBytes::<T>::insert(&bounded_id, PreUpgradeBytesWrapper(bytes));
```

Comment:
This part is a but ugly - since it puts the intermediary state into storage. But i dont think there is a spot where we can put intermediary data that is persisted between blocks.  
The threadlocal stuff gets reset after each block i think...